<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>983</width>
    <height>589</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QWidget" name="controls" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout" stretch="0">
       <property name="sizeConstraint">
        <enum>QLayout::SetMinAndMaxSize</enum>
       </property>
       <property name="leftMargin">
        <number>10</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QGroupBox" name="editMeshOptions">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Edit Mesh</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QPushButton" name="addJointButton">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Add Joint</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="addMemberButton">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Ignored" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="layoutDirection">
             <enum>Qt::LeftToRight</enum>
            </property>
            <property name="autoFillBackground">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Add Member</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="addForceButton">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Add Force</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="addSupportButton">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Add Support</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QTabWidget" name="tabWidget">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>500</width>
         <height>500</height>
        </size>
       </property>
       <property name="mouseTracking">
        <bool>true</bool>
       </property>
       <property name="autoFillBackground">
        <bool>false</bool>
       </property>
       <property name="tabShape">
        <enum>QTabWidget::Rounded</enum>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <property name="elideMode">
        <enum>Qt::ElideLeft</enum>
       </property>
       <property name="usesScrollButtons">
        <bool>false</bool>
       </property>
       <property name="documentMode">
        <bool>true</bool>
       </property>
       <property name="tabsClosable">
        <bool>true</bool>
       </property>
       <property name="movable">
        <bool>true</bool>
       </property>
       <property name="tabBarAutoHide">
        <bool>false</bool>
       </property>
       <widget class="TrussWidget" name="truss1">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <attribute name="title">
         <string>Start</string>
        </attribute>
       </widget>
       <widget class="TrussWidget" name="truss">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <attribute name="title">
         <string>New Truss</string>
        </attribute>
       </widget>
      </widget>
      <widget class="QWidget" name="infoWidget" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <property name="leftMargin">
         <number>10</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QTableWidget" name="jointInfo">
          <column>
           <property name="text">
            <string>Joint</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>X</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Y</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Movable</string>
           </property>
          </column>
         </widget>
        </item>
        <item>
         <widget class="QTableWidget" name="memberInfo">
          <column>
           <property name="text">
            <string>Member</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Joint A</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Joint B</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Internal Force</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Force Type</string>
           </property>
          </column>
         </widget>
        </item>
        <item>
         <widget class="QTableWidget" name="forceInfo">
          <column>
           <property name="text">
            <string>Force</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Joint</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>X Component</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Y Component</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Type</string>
           </property>
          </column>
         </widget>
        </item>
        <item>
         <widget class="QTableWidget" name="supportInfo">
          <column>
           <property name="text">
            <string>Support</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Joint</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Base</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>X Reaction</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Y  Reaction</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Moment Reaction</string>
           </property>
          </column>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>983</width>
     <height>37</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="separator"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_As"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <widget class="QMenu" name="menuDelete">
     <property name="title">
      <string>Delete</string>
     </property>
     <addaction name="actionJoint"/>
     <addaction name="actionMember"/>
     <addaction name="actionForce"/>
     <addaction name="actionSupport"/>
    </widget>
    <addaction name="actionedit"/>
    <addaction name="actionRedo"/>
    <addaction name="separator"/>
    <addaction name="menuDelete"/>
   </widget>
   <widget class="QMenu" name="menuSolve">
    <property name="title">
     <string>Solve</string>
    </property>
    <addaction name="actionSolve_Reactions"/>
    <addaction name="actionSolve_Members"/>
    <addaction name="separator"/>
    <addaction name="actionClear_Reactions"/>
    <addaction name="actionClear_Members"/>
    <addaction name="separator"/>
    <addaction name="actionOpen_Optimizer"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionView_in_MPL"/>
    <addaction name="separator"/>
    <addaction name="actionTruss_Preferences"/>
    <addaction name="separator"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuSolve"/>
   <addaction name="menuView"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionSave">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
   <property name="shortcutVisibleInContextMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionedit">
   <property name="text">
    <string>Undo</string>
   </property>
  </action>
  <action name="actionOpen_Optimizer">
   <property name="text">
    <string>Open Optimizer...</string>
   </property>
  </action>
  <action name="actionView_in_MPL">
   <property name="text">
    <string>View in MPL</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="text">
    <string>New Truss</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Open...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="text">
    <string>Save As...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionSolve_Reactions">
   <property name="text">
    <string>Solve Reactions</string>
   </property>
  </action>
  <action name="actionSolve_Members">
   <property name="text">
    <string>Solve Members</string>
   </property>
  </action>
  <action name="actionOptimize">
   <property name="text">
    <string>Optimize</string>
   </property>
  </action>
  <action name="actionTruss_Preferences">
   <property name="text">
    <string>Truss Preferences</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="text">
    <string>Redo</string>
   </property>
  </action>
  <action name="actionOpen_Change_Log">
   <property name="text">
    <string>Open Change Log</string>
   </property>
  </action>
  <action name="actionClear_Reactions">
   <property name="text">
    <string>Clear Reactions</string>
   </property>
  </action>
  <action name="actionClear_Members">
   <property name="text">
    <string>Clear Members</string>
   </property>
  </action>
  <action name="actionDelete_Member">
   <property name="text">
    <string>Delete Member...</string>
   </property>
  </action>
  <action name="actionDelete_Force">
   <property name="text">
    <string>Delete Force...</string>
   </property>
  </action>
  <action name="actionJoint">
   <property name="text">
    <string>Joint...</string>
   </property>
  </action>
  <action name="actionMember">
   <property name="text">
    <string>Member...</string>
   </property>
  </action>
  <action name="actionForce">
   <property name="text">
    <string>Force...</string>
   </property>
  </action>
  <action name="actionSupport">
   <property name="text">
    <string>Support...</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>TrussWidget</class>
   <extends>QWidget</extends>
   <header>trusseditor/trusswidget2.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
